src/app/modules/analytics-config/components/metrics-table-data/metrics-table-data.component.ts

```diff
@@ -0,0 +1,155 @@
+/*
+Licensed to Gisaïa under one or more contributor
+license agreements. See the NOTICE.txt file distributed with
+this work for additional information regarding copyright
+ownership. Gisaïa licenses this file to you under
+the Apache License, Version 2.0 (the "License"); you may
+not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+*/
+import { CdkDragDrop, moveItemInArray } from '@angular/cdk/drag-drop';
+import { Component, Input, OnDestroy, OnInit, Output, ViewChild } from '@angular/core';
+import { FormArray } from '@angular/forms';
+import { Subject, Subscription } from 'rxjs';
+import { AddSubtableDialogComponent } from '../add-subtable-dialog/add-subtable-dialog.component';
+import { MatDialog } from '@angular/material/dialog';
+import { MetricsTableSortConfig } from '@services/main-form-manager/models-config';
+import { MainFormService } from '@services/main-form/main-form.service';
+
+@Component({
+  selector: 'arlas-metrics-table-data',
+  templateUrl: './metrics-table-data.component.html',
+  styleUrls: ['./metrics-table-data.component.scss']
+})
+export class MetricsTableDataComponent implements OnInit, OnDestroy {
+
+  @Input() public control: FormArray = new FormArray([]);
+  @Input() public collection: string;
+  @Input() public displaySchema = true;
+  @Output() public sort = new Subject<any>();
+  @ViewChild('subTables', { static: true }) public subTables;
+
+  public dragDisabled = true;
+  public displayedColumns: string[] = ['drag', 'collection', 'field', 'columns', 'actions'];
+  private afterClosedAddSub: Subscription;
+  private metricsTableSortConfig: MetricsTableSortConfig = {};
+
+  public constructor(private dialog: MatDialog, private main: MainFormService,
+
+  ) { }
+
+
+
+  public ngOnInit(): void {
+    if (!this.collection) {
+      this.collection = this.main.getMainCollection();
+    }
+  }
+
+  public ngOnDestroy(): void {
+    if (this.afterClosedAddSub) {
+      this.afterClosedAddSub.unsubscribe();
```

Comment:
I feel like it will only unsubscribe the last subscribe that was created since they are squished. If you open twice a popup, it will just replace the subscription